crates/bevy_render/src/world_sync.rs

```diff
@@ -0,0 +1,124 @@
+use std::marker::PhantomData;
+
+use bevy_app::Plugin;
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::{
+    bundle::Bundle,
+    component::Component,
+    entity::Entity,
+    observer::Trigger,
+    query::With,
+    reflect::ReflectComponent,
+    system::{Local, Query, ResMut, Resource, SystemState},
+    world::{Mut, OnAdd, OnRemove, World},
+};
+use bevy_hierarchy::DespawnRecursiveExt;
+use bevy_reflect::Reflect;
+
+/// Marker component that indicates that its entity needs to be Synchronized to the render world
+#[derive(Component, Clone, Debug, Default, Reflect)]
+#[reflect[Component]]
+pub struct ToRenderWorld;
+
+#[derive(Component, Deref, Clone, Debug, Copy)]
+pub struct RenderEntity(Entity);
+impl RenderEntity {
+    #[inline]
+    pub fn id(&self) -> Entity {
+        self.0
+    }
+}
+
+#[derive(Component, Deref, Clone, Debug)]
+pub struct MainEntity(Entity);
+impl MainEntity {
+    #[inline]
+    pub fn id(&self) -> Entity {
+        self.0
+    }
+}
+
+// marker component that indicates that its entity needs to be despawned at the end of every frame.
+#[derive(Component, Clone, Debug, Default, Reflect)]
+pub struct RenderFlyEntity;
+
+pub(crate) enum EntityRecord {
+    // main
+    Added(Entity),
+    // render
```

Comment:
```suggestion
    // When an entity is despawned on the main world, notify the render world so that the corresponding entity can be despawned. This contains the render world entity.
```